Logging cleanup

garlick opened the pull request:
This is some minor clean-up to logging functions.

The prototypish libutil/log.c is stripped down to the bare minimum and functions were renamed to have a `log_` prefix.  I stopped short of converting all the `err()` and `err_exit()` users, so left the old names still working for those.

Adopt some RFC 5424 logging conventions in `flux_log()`, including ISO 8601 date strings, procid, and appname.

garlick commented:
Just forced a push of this PR with some additional work converting the logging internals to use RFC 5242 log messages.  I think this will be a bit lighter on the heap than the JSON based stuff it replaces.

grondo commented:
This looks great!

One comment, should we transition to `clock_gettime` instead of `gettimeofday` since POSIX [documentation](http://pubs.opengroup.org/onlinepubs/9699919799/functions/gettimeofday.html) of gettimeofday() says `"Applications should use the clock_gettime() function instead of the obsolescent gettimeofday() function."`

Also, should `wallclock_get_zulu` be included in `stdlog.h` since it is for getting the "standard log" timestamp? Or were you thinking we could expand `wallclock.h` with other wall-clock time related convenience functions?

garlick commented:
Ah I didn't know that about gettimeofday!  I will fix that.

> Or were you thinking we could expand wallclock.h with other wall-clock time related convenience functions?

I just vaguely thought `wallclock_get_zulu()` might be useful somewhere besides the logging subsystem.  My understanding is that the linker pulls in entire .o's when it finds an unresolved symbol in a static library.  If that's still the case, and stdlog.o and wallclock.o might have different users, then maybe there is an advantage to keeping them as separate libutil.a objects?

garlick commented:
One other long standing bit of (mindless) cleanup I should probably add to this PR is to add a `flux_strerror()` function that maps errno to a string, both POSIX values and zeromq values.

There are many places in our code (a quick grep says around 300) where `strerror()` is called to decode errors that might include errnos in the zeromq space.  I'll plan to replace those with `flux_strerror()` or if appropriate, `flux_log_error()` or `FLUX_LOG_ERROR()`.

grondo commented:
@SteVwonder's question in this morning's meeting reminded me that it would be nice if there was a `log-level` tunable at which log messages are discarded instead of being added to the dmesg buffer. The specific issue I'd like to solve is error messages being quickly dropped from the buffer due to lots of debug messages (e.g. when running a lot of jobs).

garlick commented:
Reasonable - I'm on it.

garlick commented:
OK, I rebased on current master, added the requested `log-level` attribute, and finished converting all the `err()` and `err_exit()` to `log_err()` and `log_err_exit()`.

Then I changed internal logging in wreck from `log_` to `wlog_` and broker log buffer class from `log_` to `logbuf_`.

garlick commented:
OK, one more change here:  move the `oom()` macro out of log.h into its own header.   There is no reason to have it be dependent on `log_errn_exit()` - it can call `fprintf(stderr)` and `exit(1)`.

garlick commented:
I'm all done poking at this now, I think.

garlick commented:
Rebased again after faketime fix landed on master.

grondo commented:
This looks good to me. Hate to ask this, but want to squash some of the intermediate commits, like whitespace and clang error fixups?

garlick commented:
There, how's that?

garlick commented:
Hmm, travis (coverage) got stuck here - I'll restart
```
PASS: t0015-cron.t
ok 1 - libfaketime works
ok 2 - load cron module
ok 3 - flux-cron tab works for set minute
not ok 4 - flux-cron tab works for any day midnight
#	
#	    id=$(echo "0 0 * * * flux event pub t.cron.complete" | flux_cron tab) &&
#	    next=$(date +%s --date="tomorrow 00:00") &&
#	    cron_entry_check ${id} type datetime &&
#	    cron_entry_check ${id} stopped false &&
#	    cron_entry_check ${id} typedata.next_wakeup ${next} &&
#	    ${event_trace} t.cron t.cron.complete \
#	        $set_faketime tomorrow 00:00 &&
#	    cron_entry_check ${id} stats.count 1 &&
#	    flux cron delete ${id}
#	
No output has been received in the last 10 minutes, this potentially indicates a stalled build or something wrong with the build itself.
The build has been terminated
```

grondo commented:
Ah, bummer. I hope those new cron tests aren't unreliable. :-(

This looks great now, ok to merge?

garlick commented:
Yep!
